common/src/main/java/org/opensearch/ml/common/connector/template/ConnectorState.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.connector.template;
+
+public enum ConnectorState {
```

Comment:
Why do we need to track the connector state as we already have model state ?

Reply:
Maybe we can reserve this connector state for future "metadata" or "healthcheck" APIs? Once the connector is not healthy, we can mark it as unhealthy? Let's keep it for a while. If we figure that it's not useful at all, we can always delete it.